html_css/v2/main-course/intermediate_css/frameworks-preprocessors.md

```diff
@@ -0,0 +1,91 @@
+### Introduction
+
+At this point, you have written quite a bit of vanilla HTML and CSS, and learned many design techniques that you will continue to use as you grow as a developer. However, through your experiences, you have no doubt encountered a CSS framework or two. You may have also learned about preprocessors (aka precompilers). Both of these types of tools can make writing CSS more streamlined and less tedious.
+
+You should be aware that at this point in your learning, both frameworks and preprocessors are options for you, but neither of them are necessary. In particular, you'll read a strong argument below AGAINST using frameworks at this point.
+
+### Learning Outcomes
+
+- What are CSS frameworks?
+- How do frameworks actually work to get you the look you want?
+- How do you interact with the framework to get the look you want?
+- What are Bootstrap, Foundation, Tailwind and Bulma?
+- What other frameworks are find out there?
+- What are the disadvantages of using frameworks?
+
+- What is a preprocessor?
+- What are the advantages of using a preprocessor?
+- What does it mean that a preprocessor "compiles to CSS"?
+- What is SASS/SCSS?
+- How can you use SASS to nest CSS?
+- How can you use SASS to work with variables in your code?
+- What are the advantages of using mix-ins with SASS?
+- If you're not using Rails, how do you set up SASS?
+- What is LESS and how is it different from SASS?
+- What other preprocessors are out there?
```

Comment:
For the purposes of TOP, I really don't think this is compatible with our idea of "use vanilla CSS until you are REALLY good at it." At this point, I don't have confidence the people that would reach for these will be at the level they should be to reach for them.

Having awareness of *what* a preprocessor and a framework is, could be important, but it's not something we want the learners to be doing a deep dive on.

Reply:
OK, these were actually lifted verbatim from the existing lesson on preprocessors. I will remove them no problem.

Reply:
ah that makes sense... we've been kind of redefining what belongs in the "learning outcomes" section of our lessons.